kore/src/Kore/Step/Simplification/AndTerms.hs

```diff
@@ -716,10 +717,20 @@ functionAnd first second
     & pure
   | otherwise = empty
   where
-    (first', second')
-      | isConstructorLike first = (first, second)
-      | isConstructorLike second = (second, first)
-      | otherwise = minMax first second
+    (first', second') =
+        case compareForEquals first second of
+            LT -> (first, second)
+            _  -> (second, first)
+
```

Comment:
Can this be factored out? It seems to be the same logic applied both here and below.

Reply:
The result of the comparison is also needed [here](https://github.com/kframework/kore/pull/2165/files#diff-ad8c8821fcc726f3ca662ce3f975422fR171), but not its arguments, so I thought it's better to have `compareForEquals` compute the `Ordering` and let each function use it the way it needs.

Reply:
Please add a function `minMaxBy :: (a -> a -> Ordering) -> (a, a) -> (a, a)` to `Prelude.Kore` like [`sortBy`](https://hackage.haskell.org/package/base-4.14.0.0/docs/Data-List.html#v:sortBy). Here we can have `minMaxBy compareForEquals ...` and below we will have `minMaxBy (on compareForEquals OrPattern.toTermLike) ...`.